langchain/src/vectorstores/vectara.ts

```diff
@@ -0,0 +1,213 @@
+  private customer_id: number;
+
+  constructor(args: VectaraLibArgs) {
+    super(new OpenAIEmbeddings(), args);
+
+    const apiKey = args.api_key ?? getEnvironmentVariable("VECTARA_API_KEY");
+    if (!apiKey) {
+      throw new Error("Vectara api key is not provided.");
+    }
+    this.api_key = apiKey;
+
+    const corpusId =
+      args.corpus_id ?? getEnvironmentVariable("VECTARA_CORPUS_ID");
+    if (!corpusId) {
+      throw new Error("Vectara corpus id is not provided.");
+    }
+    this.corpus_id = corpusId;
+
+    const customerId =
+      args.customer_id ?? getEnvironmentVariable("VECTARA_CUSTOMER_ID");
+    if (!customerId) {
+      throw new Error("Vectara customer id is not provided.");
+    }
+    this.customer_id = customerId;
+  }
+
+  async getHeader(): Promise<VectaraCallHeader> {
+    return {
+      headers: {
+        "x-api-key": this.api_key,
+        "Content-Type": "application/json",
+        "customer-id": this.customer_id.toString(),
+      },
+    };
+  }
+
+  async addVectors(
+    _vectors: number[][],
+    _documents: Document<Record<string, any>>[]
+  ): Promise<void> {
+    throw new Error(
+      "Method not implemented. Please call addDocuments instead."
+    );
+  }
+
+  async addDocuments(
+    documents: Document<Record<string, any>>[]
+  ): Promise<any> {
+    const headers = await this.getHeader();
+    let countAdded: number = 0;
+    for (const document of documents) {
+      const data = {
+        customer_id: this.customer_id,
+        corpus_id: this.corpus_id,
+        document: {
+          document_id: document.metadata.document_id,
+          title: document.metadata.title ?? "",
+          metadata_json: JSON.stringify(document.metadata),
+          section: [
+            {
+              text: document.pageContent,
+            } as any,
+          ],
+        },
+      };
+
+      try {
+        const response = await fetch(`https://${this.api_endpoint}/v1/index`, {
+          method: "POST",
+          headers: headers?.headers,
+          body: JSON.stringify(data),
+        });
+        const result = await response.json();
+        if (result.status.code !== 'OK' && result.status.code !== 'ALREADY_EXISTS') {
+          return {
+            "code": 500,
+            "detail": `Vectara API returned status code ${response.status}`,
+          }
+        } else {
+          countAdded += 1;
+        }
+      }
+      catch (e) {
+        return {
+          "code": 500,
+          "detail": `Error ${e} while adding document ${document}`,
+        };
+      }
+    }
+    return {
+      "code": 200,
+      "detail": `Added ${countAdded} documents to Vectara`,
+    };
+  }
+
+  async similaritySearchWithScore(
+    query: string,
+    _k = 4,
+    filter: this["FilterType"] | undefined = undefined
+  ): Promise<[Document, number][]> {
+    const headers = await this.getHeader();
+    const data = {
+      query: [
+        {
+          query: query,
+          numResults: 10,
```

Comment:
Seems unusual to hardcode this. Shouldn't the consumer have a way of defining this?

Reply:
The signature of the method doesn't let us do that unfortunately. 
We would like to give control to the users to specify some other thing as well, such as lambda, reranker, number of results. But with langchain, I guess we cannot do this as of now.

Reply:
AS per similar comment below, numResults should be equivalent to "_k" no?